cap/modules/importer/github_importer.py

```diff
@@ -1,24 +1,19 @@
-import os
-import urllib
-import utils
+from urllib2 import urlopen
 from github import Github
-from importer import Importer
+from base_importer import BaseImporter
 
 
-class GithubImporter(Importer):
+class GithubImporter(BaseImporter):
 
-    def __init__(self, repo, token=None):
+    def __init__(self, repo, ref=None, token=None):
         self.repo = repo
         self.token = token
+        self.ref = ref
 
     def archive_repository(self):
         """ download and archive repo via PyGithub """
         gh = Github(self.token)
-
         # self.repo format: username/repository
         repo = gh.get_repo(self.repo)
-        link = repo.get_archive_link("tarball")
-        urllib.urlretrieve(link, os.path.join(os.getcwd(), "archive.tar.gz"))
-
-    def parse_url(self, url):
-        utils.parse_urls(url)
+        link = repo.get_archive_link("tarball", ref=self.ref)
```

Comment:
Passing ```ref``` when equals to ```None``` in  in Github package ```get_archive_link``` throughs an assertion error

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/pamfilos/.envs/cap2/src/analysispreservation.cern.ch/cap/modules/importer/github_importer.py", line 18, in archive_repository
    link = repo.get_archive_link("tarball", ref=self.ref)
  File "/Users/pamfilos/.envs/cap2/lib/python2.7/site-packages/github/Repository.py", line 1085, in get_archive_link
    assert ref is github.GithubObject.NotSet or isinstance(ref, (str, unicode)), ref
AssertionError: None
```

Reply:
Fixed now